extras/upgrade_activities.sh

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2018 Red Hat, Inc. <http://www.redhat.com>
+# This file is part of gluster-block.
+#
+# This file is licensed to you under your choice of the GNU Lesser
+# General Public License, version 3 or any later version (LGPLv3 or
+# later), or the GNU General Public License, version 2 (GPLv2), in all
+# cases as published by the Free Software Foundation.
+
+# This script is meant for running upgrade activities,
+# it is triggered by gluster-blockd unit startup.
+
+GB_STATUSFILE="/var/lib/gluster-block/gb_upgrade.status"
+LOGDIR="${GB_LOGDIR:-/var/log/gluster-block}"
+GB_LOGFILE="${LOGDIR}/gluster-block-upgrade-activities.log"
+GB_SAVEFILE="/etc/target/saveconfig.json"
+GB_TMP_SAVEFILE="/tmp/gb_saveconfig.json"
+USED_COMMANDS="ps sed grep awk head echo gluster gluster-block"  # append new use of commands here
+mkdir -p "${LOGDIR}"
+
+
+function printLog()
+{
+  local msg=${1}
+
+  echo "[$(date -u +'%Y-%m-%d %I:%M:%S')] ${msg}" >> ${GB_LOGFILE}
+}
+
+
+function get_bhv_list()
+{
+  local status=${1}
+
+  gluster volume info | \
+  # get volume name, status, shard-status, number-of-bricks
+  grep -P "(^Volume Name:|^Status:|^features.shard:|^Number of Bricks:)" | \
+  # get the volumes based on status
+  sed '/^Volume Name:/{x;p;x;}' | sed -e '/./{H;$!d;}' -e "x;/Status: ${status}/!d;" | \
+  # get shard enabled volumes
+  sed -e '/./{H;$!d;}' -e 'x;/features.shard: on/b' -e '/features.shard: yes/b' -e '/features.shard: enable/b' -e '/features.shard: true/b' -e '/features.shard: 1/b' -e d | \
+  # get comma separated volume names
+  grep -P "(^Volume Name:)" | awk '{print $NF}' | sed -e '{:a;N;$!ba};s/\n/,/g'
+}
+
+
+function get_local_hostip()
+{
+  local ipaddr
+
+  while read -r ipaddr
+  do
+    if [[ $(ip addr | grep -c ${ipaddr}) -eq 1 ]]
+    then
+      echo ${ipaddr}
+      break
+    fi
+  done < <(gluster volume info | grep -P "(^Brick[0-9]*:)" | awk -F':' '{print $2;}')
+}
+
+
+function check_daemons_running()
+{
+  if ! ps cax | grep -wq '[g]lusterd'
+  then
+    printLog "ERROR: Glusterd is not running"
+    exit 1
+  fi
+
+  if ! ps cax | grep -wq '[g]luster-blockd'
```

Comment:
@pkalever This is not working.

 18 [2018-10-15 07:05:55] INFO: skipping upgrade activities
 19 [2018-10-15 07:08:28] INFO: started upgrade activities
 20 [2018-10-15 07:08:28] ERROR: Gluster-blockd is not running
 21 [2018-10-15 07:11:00] INFO: started upgrade activities
 22 [2018-10-15 07:11:01] INFO: no block hosting volumes found
 23 [2018-10-15 07:11:01] INFO: successfully completed the upgrade activities


It will work by:
if ! ps cax | grep -q '[g]luster-blockd'
or
if ! ps cax | grep -wq '[g]luster-blockd'  > /dev/null 2>&1